extensions/replication/queueProcessor/QueueProcessor.js

```diff
@@ -66,8 +67,31 @@ class _UserAuthManager {
 }
 
 class _RoleAuthManager {
-    constructor() {
-        throw Error('Role-based authentication not yet implemented');
+    constructor(authConfig, log) {
+        this._log = log;
+        const { host, port } = authConfig.vault;
+        this._vaultclient = new Vaultclient(host, port);
+    }
+
+    getCredentialProvider() {
```

Comment:
This function returns `undefined`, but it may be temporary until we implement the credentials properly. (actually I don't know if `getCredentialProvider()` is going to stay with the credentials cache etc.).